syntax = "proto3";

package org.megras.util.services;

// Define the service for CLIP embeddings
service ClipService {
  // Method to get text embeddings
  rpc GetTextEmbedding (TextRequest) returns (EmbeddingResponse);

  // Method to get image embeddings
  rpc GetImageEmbedding (ImageRequest) returns (EmbeddingResponse);
}

// Message for text embedding requests
message TextRequest {
  string text = 1;
}

// Message for image embedding requests
message ImageRequest {
  bytes image_data = 1; // Now sends the raw bytes of the image
}

// Message for embedding responses (for both text and image)
message EmbeddingResponse {
  repeated float embedding = 1; // CLIP embeddings are typically float arrays
}